vllm/v1/engine/core_client.py

```diff
@@ -176,7 +188,20 @@ async def get_output_async(self) -> List[EngineCoreOutput]:
 
         return engine_core_outputs
 
+    async def send_input(self,
+                   request_type: EngineCoreRequestType,
+                   request: Union[EngineCoreRequest, EngineCoreAbortRequest])\
+                           -> None:
+        await self.input_socket.send_multipart((
+            self.encoder.encode(request_type),
+            self.encoder.encode(request),
+        ),
```

Comment:
@robertgshaw2-neuralmagic -  @njhill  and I spoke about using a single socket so the poll doesn't become a bottleneck. as part of it, we send both the "Request Type" and "Request Data" - the "Request Type" is used to determine what appropriate decoder type in the LLMEngineCore. What do you think ?

Reply:
I like this design